File: lib/python/aresdoc/to_snake.py
# https://github.com/Zeronetsec/Ares

import re

def execute(name: str) -> str:
    return re.sub(
        r"([a-z0-9])([A-Z])",
        r"\1_\2",
        name,
    ).lower()

# Copyright (c) 2026 Zeronetsec

File: bin/core/aresdoc/to_snake.go
// https://github.com/Zeronetsec/Ares

package main

import (
    "strings"
)

func toSnake(name string) string {
    res := camelCaseRe.ReplaceAllString(
        name,
        "${1}_${2}",
    )
    return strings.ToLower(res)
}

// Copyright (c) 2026 Zeronetsec